components/script/dom/gputexture.rs

```diff
@@ -0,0 +1,191 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::cell::DomRefCell;
+use crate::dom::bindings::codegen::Bindings::GPUTextureBinding::{
+    GPUExtent3DDict, GPUTextureDimension, GPUTextureFormat, GPUTextureMethods,
+};
+use crate::dom::bindings::codegen::Bindings::GPUTextureViewBinding::{
+    GPUTextureAspect, GPUTextureViewDescriptor,
+};
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject, Reflector};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::bindings::str::DOMString;
+use crate::dom::globalscope::GlobalScope;
+use crate::dom::gpudevice::{convert_texture_format, convert_texture_view_dimension};
+use crate::dom::gputextureview::GPUTextureView;
+use dom_struct::dom_struct;
+use std::cell::Cell;
+use webgpu::{wgt, WebGPU, WebGPUDevice, WebGPURequest, WebGPUTexture, WebGPUTextureView};
+
+#[dom_struct]
+pub struct GPUTexture {
+    reflector_: Reflector,
+    texture: WebGPUTexture,
+    label: DomRefCell<Option<DOMString>>,
```

Comment:
would it make sense to just put the whole GPUTextureDescriptor in here instead of all the fields separately?

Reply:
I don't think it would do much benefit. Instead, it would make things a little hard to access- eg the label in the descriptor and size field. We'll have to use `match` everytime we need to access texture_size (like while creating `TextureView`). Currently I am storing `GPUExtent3DDict` instead of `GPUExtent3D` which makes it easier to access. The label is stored inside parent field in the descriptor as `Option<Option<DOMString>>`.